WordPress/Classes/ViewRelated/NUX/WordPressAuthenticationManager.swift

```diff
@@ -366,7 +366,13 @@ extension WordPressAuthenticationManager: WordPressAuthenticatorDelegate {
             // If the quick start prompt has already been dismissed,
             // then show the My Site screen for the specified blog
             guard !self.quickStartSettings.promptWasDismissed(for: blog) else {
-                self.windowManager.dismissFullscreenSignIn(blogToShow: blog)
+
+                if self.windowManager.isShowingFullscreenSignIn {
+                    self.windowManager.dismissFullscreenSignIn(blogToShow: blog)
+                } else {
+                    navigationController.dismiss(animated: true)
+                }
```

Comment:
@leandroalonso 
This accounts for the case when we're already logged in, and we add a self-hosted site via the site switcher.
The UI test below caught this case -- phew!
https://github.com/wordpress-mobile/WordPress-iOS/blob/3675ad4ed9bc44dd374a2b2b10c1fdf6efb587d1/WordPress/WordPressUITests/Tests/LoginTests.swift#L73-L75